common/cpp/tests/Makefile

```diff
@@ -12,10 +12,19 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-CPPFLAGS= -D_UNTRUSTED_
+CPPFLAGS= -D_UNTRUSTED_ -DCRYPTOLIB_OPENSSL
 CPPFLAGS+= -I..  -I../crypto -I../packages/base64
 LDFLAGS=-lcrypto
-UTILTESTOBJS= build/crypto_utils.o build/skenc.o build/types.o \
+
+# To use Mbed TLS instead of the default OpenSSL, define CRYPTOLIB_MBEDTLS=1
+# For example: CRYPTOLIB_MBEDTLS=1 make
+ifdef CRYPTOLIB_MBEDTLS
+	CPPFLAGS+= -DCRYPTOLIB_MBEDTLS
+	LDFLAGS+=-lmbedcrypto
```

Comment:
Don't we need conditional compilation for `CPPFLAGS` for OPENSSL as well. `CPPFLAGS= -D_UNTRUSTED_ -DCRYPTOLIB_OPENSSL` is always added. So if `CRYPTOLIB_MBEDTLS` is defined we will have both OPENSSL and MBEDTLS

Reply:
@manojgop You are correct--we will need the same for OpenSSL as well. In the transition period, only part of the crypto functions are implemented with Mbed TLS--the rest is OpenSSL.  In other words, there are two choices
- Pure OpenSSL
- Mixed Mbed TLS + OpenSSL

In the future it will be:
- Pure Openssl
- Pure Mbed TLS
But we are not there yet.